content/03-types/02-relational/07-what-are-joins.mdx

```diff
@@ -0,0 +1,401 @@
+
+> **Note:** `LATERAL JOIN`s are supported by Postgres but only supported in [MySQL versions >8](https://dev.mysql.com/blog-archive/support-for-lateral-derived-tables-added-to-mysql-8-0-14/)
+
+In other words, a `LATERAL JOIN` is like a foreach loop, where PostgreSQL iterates over each row in a result set and uses each row to evaluate the subquery.
+
+Here is the associated query for the `customers` and `orders` table:
+
+```sql
+SELECT
+    *
+FROM
+    customers
+	LEFT JOIN LATERAL
+	    (SELECT *
+       FROM orders
+       WHERE orders.customer_id = customers.customer_id
+       ) AS alias ON TRUE;
+
+```
+
+Here is the result table after using a `LATERAL JOIN`:
+
+| customer_id | name    | age | email address     | order_id | cost    | order_date |
+| ----------- | ------- | --- | ----------------- | -------- | ------- | ---------- |
+| 1           | Jenny   | 30  | jenny@prisma.io   | 1        | $150.00 | 4/1/2024   |
+| 1           | Jenny   | 30  | jenny@prisma.io   | 3        | $20.00  | 4/2/2024   |
+| 2           | Adrian  | 28  | adrian@prisma.io  | 2        | $200.00 | 3/29/2024  |
+| 3           | Synniva | 25  | synniva@prisma.io | NULL     | NULL    | NULL       |
+
+In this case, the result would be the same as a `LEFT JOIN` and `FULL JOIN`.
+
+<PrismaOutlinks>
+
+💡 Prisma ORM’s preview JOIN feature works with Postgres and MySQL versions >8. To read more about the preview feature, read the blog post [here](https://www.prisma.io/blog/prisma-orm-now-lets-you-choose-the-best-join-strategy-preview)
+
+</PrismaOutlinks>
+
+## Implementing JOINs for different relations
+
+It’s important to understand database relations when selecting the appropriate JOIN type. In the context of databases, relations describe a relationship between tables in a database. 
+
+In this section, we will illustrate both the one-to-one and the one-to-many relations along with considerations for selecting the appropriate JOIN type for each.
+
+
+### One-to-one (1-1)
+
+In a one-to-one relation, each record in one table is associated with exactly one record in another table, and vice versa. Here is an example SQL schema demonstrating a one-to-one relation between `User` and `Profile` models:
+
+```sql
+CREATE TABLE "User" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "email" TEXT NOT NULL
+);
+
+CREATE TABLE "Profile" (
+    "name" TEXT NOT NULL,
+    "gender" TEXT NOT NULL,
+    "age" TEXT NOT NULL,
+    "userEmail" TEXT NOT NULL,
+    CONSTRAINT "Profile_userEmail_fkey" FOREIGN KEY ("userEmail") REFERENCES "User" ("email") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+CREATE UNIQUE INDEX "User_email_key" ON "User"("email");
+
+CREATE UNIQUE INDEX "Profile_userEmail_key" ON "Profile"("userEmail");
+
+```
+
+This means that a user can have only one profile, and a profile can belong to only one user.
```

Comment:
I think this snippet could be confusing because it uses a unique field as a foreign key which hasn't been explained as a possibility in this article and also isn't mentioned anywhere here as an additional explanation. A reader who is new to the topic will wonder how this is possible because all previous examples used primary keys to be mapped to foreign keys.

I suggest to either use the previous approach of using a primary key or add an additional explanation.